opm/autodiff/FlowMainEbos.hpp

```diff
@@ -805,6 +808,28 @@ namespace Opm
         Scalar gravity() const
         { return ebosProblem().gravity()[2]; }
 
+        std::map<std::string, std::vector<int> > computeCellRanks(bool output, bool output_ecl)
+        {
+            std::map<std::string, std::vector<int> > integerVectors;
+
+            if(  output && output_ecl && grid().comm().size() > 1 )
+            {
+                // Get the owner rank number for each cell
+                using ElementMapper =  Dune::MultipleCodimMultipleGeomTypeMapper<GridView, Dune::MCMGElementLayout>;
+                using Handle = CellOwnerDataHandle<ElementMapper>;
+                ElementMapper globalMapper(this->globalGrid().leafGridView());
+                const auto* dims = UgGridHelpers::cartDims(grid());
+                const auto globalSize = dims[0]*dims[1]*dims[2];
```

Comment:
This will be a significant extra overhead for many real cases, now that the upstream API has been changed we should rather use the number of active cells.

Reply:
PR is coming. Gladly @joakim-hove merged anyway. Otherwise opm-simulator would be broken as OPM/opm-output#216 was not backwards compatible.